client/state/ui/editor/media/reducer.js

```diff
@@ -17,6 +32,22 @@ export function editItem( state = null, action ) {
 	return state;
 }
 
+export function imageEditor( state = defaultImageEditorState, action ) {
+	switch ( action.type ) {
+		case IMAGE_EDITOR_ROTATE:
+			return Object.assign( {}, state, { changed: true, rotate: action.degrees } );
+		case IMAGE_EDITOR_SCALE:
+			return Object.assign( {}, state, { changed: true, scaleX: action.scaleX, scaleY: action.scaleY } );
+		case IMAGE_EDITOR_SET_FILE_INFO:
+			return Object.assign( {}, state, { src: action.src, fileName: action.fileName } );
+		case IMAGE_EDITOR_STATE_RESET:
+			return Object.assign( {}, state, { changed: false, rotate: 0, scaleX: 1, scaleY: 1 } );
```

Comment:
Maybe `hasChanges` instead of `changed`? It looks like this is backing the reset button. 

Optional here, but it might be nice to pull out this changed bit from the canvas info:
```
imageEditor: {
     hasChanges: true,
     src: src,
    transform: { scaleX, scaleY, degrees }
}
```

Where the hasChanges reducer might look like:
```
export function hasChanges( state = false, action ) {
  switch ( action.type ) {
    case IMAGE_EDITOR_ROTATE_COUNTERCLOCKWISE:
    case IMAGE_EDITOR_FLIP:
    case IMAGE_EDITOR_STATE_RESET:
         return action.type !== IMAGE_EDITOR_STATE_RESET
}
return state;
```